Add XML syntax for configuring SSL

tokle opened the pull request.

bjorncs committed:
Add interface for SslContextFactoryProvider

bjorncs committed:
Move logic for creating default SslContextFactory to DefaultSslContextFactoryProvider

Commit:
Add ssl and ssl-provider elements

Commit:
Set up components and config for ssl/ssl-provider

bjorncs committed:
Add dummy implementation of SslContextFactoryProvider

bjorncs committed:
Inject ThrowingSslContextFactoryProvider for non-ssl connectors

bjorncs commented:
@gjoranv Ready for review

gjoranv commented on jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/ThrowingSslContextFactoryProvider.java:
```diff
@@ -0,0 +1,16 @@
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+
+/**
+ * A dummy implementation of {@link SslContextFactoryProvider} to be injected into non-ssl connectors
```
I suggest to copy this comment into the dummy in the model as well.

bjorncs committed:
Misc changes based on input from code review

- Use separate string for each line as input to parse()
- Improve name of unit test
- Use Class.getName() instead of hardcoded strings
- Improve Javadoc

bjorncs commented:
@gjoranv PTAL

bjorncs committed:
Sort dependencies on scope

- Also remove dependency on Jetbrains annotations artifact

bjorncs committed:
Use new ssl config to setup ssl on connector

tokle commented on config-model/src/test/java/com/yahoo/vespa/model/container/xml/JettyContainerModelBuilderTest.java:
```diff
@@ -229,6 +234,87 @@ public void ssl_keystore_and_truststore_configurator_can_be_overriden() throws I
 
+    @Test
+    public void verify_default_ssl_configuration() {
+        Element clusterElem = DomBuilderTest.parse(
```
Might look a bit better, but breaks possibility to use intellij "Inject language" feature.

Commit:
ssl-provider ComponentDefinition -> BundleSpec

gjoranv commented on jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/DefaultSslContextFactoryProvider.java:
```diff
@@ -58,20 +62,61 @@ public SslContextFactory getInstance(String containerId, int port) {
+            }
+            factory.setProtocol("TLS");
+        } else {
+            // TODO Remove SslKeyStoreConfigurator / SslTrustStoreConfigurator
```
TODO Vespa 7?

bjorncs replied:
TODO later this sprint. This was never a public api :)

gjoranv commented on jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/DefaultSslContextFactoryProvider.java:
```diff
@@ -58,20 +62,61 @@ public SslContextFactory getInstance(String containerId, int port) {
+            sslKeyStoreConfigurator.configure(new DefaultSslKeyStoreContext(factory));
+            sslTrustStoreConfigurator.configure(new DefaultSslTrustStoreContext(factory));
+
+            // TODO Remove support for deprecated ssl connector config
```
TODO Vespa 7?

bjorncs replied:
TODO later this sprint. This was never a public api :)

gjoranv replied:
Add "Vespa 7" to the comment

bjorncs commented:
Rebased with master
